src/block.js

```diff
@@ -115,4 +116,25 @@ Block.prototype.toHex = function (headersOnly) {
   return this.toBuffer(headersOnly).toString('hex')
 }
 
+Block.calculateTarget = function Target (bits) {
+  var target = new Buffer(32)
+  target.fill(0)
+
+  var exponent = ((bits & 0xff000000) >> 24) - 3
+  var value = bits & 0x00ffffff
+  var i = 31 - exponent
+  if (value & 0xff) target[i--] = (value & 0xff)
+  if (value >>> 8) target[i--] = (value >>> 8)
+  if (value >>> 16) target[i--] = (value >>> 16)
+  if (value >>> 24) target[i--] = (value >>> 24)
```

Comment:
This zero trimming is the only part I'm still not sure why it happens.  Need to verify it with more fixtures.